airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumRecordIterator.java

```diff
@@ -134,15 +134,33 @@ protected ChangeEvent<String, String> computeNext() {
 
       // if the last record matches the target file position, it is time to tell the producer to shutdown.
       if (targetPosition.reachedTargetPosition(eventAsJson)) {
-        LOGGER.info("Closing: Change event reached target position");
-        requestClose();
+        requestClose("Closing: Change event reached target position");
       }
       this.tsLastHeartbeat = null;
       this.lastHeartbeatPosition = null;
       this.receivedFirstRecord = true;
       this.maxInstanceOfNoRecordsFound = 0;
       return next;
     }
+
+    if (!signalledDebeziumEngineShutdown) {
+      LOGGER.warn("Debezium engine has not been signalled to shutdown, this is unexpected");
+    }
```

Comment:
Why is this unexpected?

Reply:
Cause we should have shutdown the engine by this point of execution. This will never happen today but it's more of a safety log that I placed to catch bugs in case of future changes to the code.